test/conformance/helpers/broker_data_plane_test_helper.go

```diff
@@ -0,0 +1,458 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingv1beta1 "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+	"knative.dev/eventing/test/lib"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/recordevents"
+	"knative.dev/eventing/test/lib/resources"
+	"knative.dev/eventing/test/lib/sender"
+
+	ce "github.com/cloudevents/sdk-go/v2"
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+)
+
+//At ingress
+//Supports CE 0.3 or CE 1.0 via HTTP
+//Supports structured or Binary mode
+//Respond with 2xx on good CE
+//Respond with 400 on bad CE
+//Reject non-POST requests to publish URI
+func BrokerV1Beta1IngressDataPlaneTestHelperWithChannelTestRunner(
+	t *testing.T,
+	brokerClass string,
+	channelTestRunner testlib.ComponentsTestRunner,
+	setupClient ...testlib.SetupClientOption,
+) {
+	channelTestRunner.RunTests(t, testlib.FeatureBasic, func(t *testing.T, channel metav1.TypeMeta) {
+		client := testlib.Setup(t, true, setupClient...)
+		defer testlib.TearDown(client)
+		brokerName := "br"
+		triggerName := "trigger"
+		loggerName := "logger-pod"
+		eventTracker, _ := recordevents.StartEventRecordOrFail(client, loggerName)
+		defer eventTracker.Cleanup()
+		client.WaitForAllTestResourcesReadyOrFail()
+
+		config := client.CreateBrokerConfigMapOrFail(brokerName, &channel)
+
+		broker := client.CreateBrokerV1Beta1OrFail(
+			brokerName,
+			resources.WithBrokerClassForBrokerV1Beta1(brokerClass),
+			resources.WithConfigForBrokerV1Beta1(config),
+		)
+		client.WaitForResourceReadyOrFail(broker.Name, testlib.BrokerTypeMeta)
+
+		trigger := client.CreateTriggerOrFailV1Beta1(
+			triggerName,
+			resources.WithBrokerV1Beta1(broker.Name),
+			resources.WithAttributesTriggerFilterV1Beta1(eventingv1beta1.TriggerAnyFilter, eventingv1beta1.TriggerAnyFilter, nil),
+			resources.WithSubscriberServiceRefForTriggerV1Beta1(loggerName),
+		)
+
+		client.WaitForResourceReadyOrFail(trigger.Name, testlib.TriggerTypeMeta)
+
+		t.Run("Ingress Supports CE0.3", func(t *testing.T) {
+			eventID := "CE0.3"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CE0.3")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+
+			if err := event.SetData(ce.ApplicationJSON, []byte(body)); err != nil {
+				t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+			}
+			event.Context.AsV03()
+			event.SetSpecVersion("0.3")
+			client.SendEventToAddressable("v03-test-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSource(event.Source()),
+				cetest.HasSpecVersion("0.3"),
```

Comment:
can you use the constant `cloudevents.SpecVersion03`?

Reply:
`% cd $GOPATH/src/github.com/cloudevents/sdk-go && rg SpecVersion03` doesn't return any results?